internal/services/network/network_connection_monitor_resource.go

```diff
@@ -175,13 +176,15 @@ func resourceNetworkConnectionMonitorSchema() map[string]*pluginsdk.Schema {
 							workspaces.ValidateWorkspaceID,
 							commonids.ValidateSubnetID,
 							commonids.ValidateVirtualNetworkID,
+							machines.ValidateMachineID,
 						),
 					},
 
 					"target_resource_type": {
 						Type:     pluginsdk.TypeString,
 						Optional: true,
 						ValidateFunc: validation.StringInSlice([]string{
+							string(connectionmonitors.EndpointTypeAzureArcVM),
```

Comment:
Is there a reason not to use `connectionmonitors.PossibleValuesForEndpointType()` here to save adding these one by one?

Reply:
Because this property still exists other properties what TF doesn't support like "AzureVMSS" and "AzureArcNetwork". As this resource isn't implemented by me, so I don't have full context about why they aren't supported. So I intend to firstly add support for the feature what user is requesting. I assume we can use connectionmonitors.PossibleValuesForEndpointType() when users request other features.